src/types/duration.js

```diff
@@ -0,0 +1,70 @@
+const ArgumentType = require('./base');
+
+class DurationArgumentType extends ArgumentType {
+	// This.duration::Milliseconds
+	constructor(client) {
+		super(client, 'duration');
+		this.timeIds = new Set(['mo', 'w', 'd', 'h', 'm', 's', 'ms']);
+		this.duration = 0;
+	}
+
+	// Confirms match before continuing.
+	// Checks if matched number is an int.
+	// Checks if matched string is in expected set.
+	validate(value) {
+		const MATCHES_ALL = value.match(/(\d+)\s*([A-Za-z]+)/g);
+
+		for(var i = 0; i < MATCHES_ALL.length; i++) {
+			var tempNum = MATCHES_ALL[i].match(/(\d+)/g);
```

Comment:
Unexpected `var`, use `const` instead.

```suggestion
			const tempNum = MATCHES_ALL[i].match(/(\d+)/g);
```